release_planning.rb

```diff
@@ -11,98 +44,99 @@
 end
 
-if due_by_commit and due_by_time
-  due_for_release = due_by_commit & due_by_time
-elsif due_by_commit
-  due_for_release = due_by_commit
-else
-  due_for_release = due_by_time
-end
+due_for_release = if due_by_commit && due_by_time
+                    due_by_commit & due_by_time
+                  elsif due_by_commit
+                    due_by_commit
+                  else
+                    due_by_time
+                  end
 
 due_for_release.each do |entry|
   puts "#{entry['repo']} is due for release. Last release was tagged on #{entry['date']} and there have been #{entry['commits']} commits since then."
 end
 
 html = []
-html.push("<html>")
-html.push("<head><script src='./web_libraries/sorttable.js'></script><link rel='stylesheet' href='./web_libraries/bootstrap.min.css'></head>")
-html.push("<body>")
-html.push("<h2>Modules Requiring Release</h2>")
-html.push("<table border='1' style='width:100%' class='sortable table table-hover'> <tr>")
-html.push("<th>Module Name</th>")
-html.push("<th>Last Release Tag Date</th>")
-html.push("<th>Commits Since Then</th>")
-html.push("</tr>")
-due_for_release.each do |entry|
-  html.push("<tr>")
-  html.push("<td>#{entry['repo']}</td>")
-  html.push("<td>#{entry['date']}</td>")
-  html.push("<td align=\"center\">#{entry['commits']}</td>")
-  html.push("</tr>")
+html.push('<html>')
+html.push("<head>")
+html.push("<script src='./web_libraries/jquery.min.js' type='text/javascript'></script>")
+html.push("<script src='./web_libraries/sorttable.js'></script><link rel='stylesheet' href='./web_libraries/bootstrap.min.css'>")
+html.push("<script src='./web_libraries/DataTables/datatables.js'></script><link rel='stylesheet' href='./web_libraries/DataTables/datatables.css'>")
+
+html.push("<script type= 'text/javascript'> $(document).ready( function () {$('#id_table').DataTable();} ); </script>")
+html.push("</head>")
+
+
+html.push('<body>')
+html.push('<h2>Modules Requiring Release</h2>')
+html.push("<table border='1' id = 'id_table' style='width:100%' class='sortable table table-hover'> <tr>")
+html.push('<th>Module Name</th>')
+html.push('<th>Last Release Tag Date</th>')
+html.push('<th>Commits Since Then</th>')
+html.push('<th>Number of downloads</th>')
+html.push('</tr>')
+repo_data.each do |puppet_module|
+  html.push('<tr>')
+  html.push("<td><a href='https://github.com/#{puppet_module['repo']}'>#{puppet_module['repo']}</a></td>")
+  html.push("<td>#{puppet_module['date']}</td>")
+  html.push("<td align=\"center\">#{puppet_module['commits']}</td>")
+  html.push("<td align=\"center\">#{puppet_module['downloads']}</td>")
+  html.push('</tr>')
 end
-html.push("</body>")
-html.push("</html>")
+html.push('</body>')
+html.push('</html>')
 
 if options[:output]
-  File.open("ModulesRelease.html", "w+") do |f|
+  File.open('ModulesRelease.html', 'w+') do |f|
     f.puts(html)
   end
 
-  File.open("ModulesRelease.json", "w") do |f|
+  File.open('ModulesRelease.json', 'w') do |f|
```

Comment:
Please open all files in binary mode (`'wb'` see https://ruby-doc.org/core-2.1.4/IO.html#method-c-new) so that the files work across all platforms.